filament/backend/src/vulkan/VulkanCommands.cpp

```diff
@@ -271,7 +271,42 @@ void VulkanCommands::updateFences() {
     }
 }
 
-} // namespace filament::backend
+bool VulkanCommands::isValidFence(VkFence inFence) const {
+    for (auto& wrapper: mStorage) {
+        if (wrapper.cmdbuffer == VK_NULL_HANDLE) {
+            continue;
+        }
+        VulkanCmdFence const* fence = wrapper.fence.get();
+        if (!fence) {
+            continue;
+        }
+        VkResult const result = fence->status.load(std::memory_order_relaxed);
+        if (fence->fence == inFence && (result == VK_INCOMPLETE || result == VK_NOT_READY)) {
+            return true;
+        }
+    }
+    return false;
+}
+
+std::vector<VkFence> VulkanCommands::getFences() const {
+    std::vector<VkFence> fences;
```

Comment:
maybe you can reserve mStorage.size() to avoid multiple allocations at push_back? Also considere using a FixedCapacityVector.